core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/ListLineConsumer.java

```diff
@@ -24,21 +24,52 @@
  */
 public class ListLineConsumer implements LineConsumer {
   protected final LinkedList<String> lines;
+  private long textSizeLimit = -1;
+  private long consumedTextSize;
 
   public ListLineConsumer() {
     lines = new LinkedList<>();
   }
 
+  /**
+   * Limit size of consuming text.
+   *
+   * @param textSizeLimit maximum number of symbols of consuming text. Don't limit data if {@code
+   *     textSizeLimit} < 0. Don't consume if {@code textSizeLimit} = 0.
+   */
+  public ListLineConsumer(long textSizeLimit) {
+    lines = new LinkedList<>();
+    this.textSizeLimit = textSizeLimit;
+  }
+
   @Override
   public void writeLine(String line) {
+    if (textSizeLimit < 0) {
+      lines.add(line);
+      return;
+    }
+
+    if (consumedTextSize >= textSizeLimit) {
+      return;
+    }
+
+    if (line != null && (line.length() + consumedTextSize > textSizeLimit)) {
```

Comment:
Does it consume only first `testSizeLimit` characters? Will it be more relevant to consume the last  `testSizeLimit` characters? I would personally introduce a new class for that.

Reply:
1. `Does it consume only first testSizeLimit characters?`
Yes

1. `Will it be more relevant to consume the last testSizeLimit characters?`
I proceeded from the fact that usually important information about an error is being placed at the start of message.

1. `I would personally introduce a new class for that.`
Agree